src/Microsoft.Health.Fhir.Core.UnitTests/Models/PartialDateTimeTests.cs

```diff
@@ -248,31 +190,32 @@ public void GivenADateImtOffset_WhenInitialized_ThenCorrectPartialDateTimeShould
             Assert.Equal(utcOffset, partialDateTime.UtcOffset);
         }
 
-        [Fact]
-        public void GivenANullString_WhenParsing_ThenExceptionShouldBeThrown()
-        {
-            Assert.Throws<ArgumentNullException>(ParamNameS, () => PartialDateTime.Parse(null));
-        }
-
         [Theory]
         [InlineData("")]
         [InlineData("    ")]
-        public void GivenAnInvalidString_WhenParsing_ThenExceptionShouldBeThrown(string s)
+        [InlineData("        ")]
+        public void GivenAnEmptyStringOrWhiteSpaces_WhenParsing_ThenArgumentExceptionShouldBeThrown(string inputString)
         {
-            Assert.Throws<ArgumentException>(ParamNameS, () => PartialDateTime.Parse(s));
+            Assert.Throws<ArgumentException>(() => PartialDateTime.Parse(inputString));
         }
 
         [Fact]
-        public void GivenAnInvalidFormatString_WhenParsing_ThenExceptionShouldBeThrown()
+        public void GivenANullString_WhenParsing_ThenArgumentExceptionShouldBeThrown()
+        {
+            Assert.Throws<ArgumentNullException>(() => PartialDateTime.Parse(null));
+        }
+
+        [Theory]
+        [InlineData("****")]
+        [InlineData("!")]
+        [InlineData("abc")]
+        public void GivenAnInvalidString_WhenParsing_ThenFormatExceptionShouldBeThrown(string inputString)
         {
-            Assert.Throws<FormatException>(() => PartialDateTime.Parse("abc"));
+            Assert.Throws<FormatException>(() => PartialDateTime.Parse(inputString));
         }
 
         public static IEnumerable<object[]> GetParseData()
         {
-            yield return new object[] { "2017", 2017, null, null, null, null, null, null, null };
-            yield return new object[] { "2017-01", 2017, 1, null, null, null, null, null, null };
-            yield return new object[] { "2019-01-02", 2019, 1, 2, null, null, null, null, null };
```

Comment:
Are these three scenarios tested elsewhere?

Reply:
Yep exactly - they are included in `GetParameterNullData()`.